mne/channels/montage.py

```diff
@@ -206,6 +206,16 @@ def read_montage(kind, ch_names=None, path=None, unit='m', transform=False):
         ch_names_ = []
         pos = []
         with open(fname) as fid:
+            # Default units are meters
+            scale_factor = 1.
+            for line in fid:
+                if 'UnitPosition' in line:
+                    units = line.split()[1]
+                    if units == 'm':
+                        break
+                    elif units == 'mm':
+                        scale_factor = 1. / 1000.
+                        break
```

Comment:
Instead of doing this, I would do:
```
scale_factor = dict(m=1., mm=1e-3)[units]
break
```
This has the advantage of failing if a file has units we don't have defined currently. Then at the end of your loop you can add:
```
else:
    raise RuntimeError('Could not detect units in file %s' % fname)
```
This has the advantage of ensuring that the file has units in it.

Better to be restrictive/safe so we don't hit this issue again without it being obvious via error.